core/app/models/spree/promotion_code.rb

```diff
@@ -0,0 +1,26 @@
+class Spree::PromotionCode < ActiveRecord::Base
+  belongs_to :promotion
+
+  validates :promotion, :value, presence: true
+  validates :usage_limit, numericality: { greater_than: 0, allow_nil: true }
+
+  before_save :normalize_value
+
+  def expired?
+    !!(starts_at && Time.now < starts_at || expires_at && Time.now > expires_at)
+  end
+
+  def eligible?(promotable)
+    !expired? && !usage_limit_exceeded?(promotable)
+  end
+
+  def usage_limit_exceeded?(promotable)
+    false
```

Comment:
I'm still not too sure how to handle this.
The promotion usage limit counts the number of adjustments for that promotion's actions, but it is no way linked to the promotion code... Any suggestions?